tools/sloop.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * Library of functions pertaining to Sloop shadows. Slooping is a way
+ * to simulate the built-in snooping functionality of EotL.
+ *
+ * @author gon
+ */
+
+#define SLOOPEE_SHAD "/usr/gon/tools/sloopee_shad"
+#define SLOOPER_SHAD "/usr/gon/tools/slooper_shad"
+
+// Function Declarations
+protected void add_sloop_shadow( object ob, string str );
+protected void dest_sloop_shadows( object ob, string str );
+protected object * query_sloop_shadows( object ob, string str );
+void add_slooper_shadow( object ob );
+void add_sloopee_shadow( object ob );
+void dest_slooper_shadow( object ob );
+void dest_sloopee_shadow( object ob );
+object find_slooper_shadow( object ob );
+object find_sloopee_shadow( object ob );
+
+/**
+ * Function filters all shdow objects and returns a filtered array of shadows
+ * matching either SLOOPEE_SHAD or SLOOPER_SHAD
+ * @param  object ob   Object passed to check for shadow
+ * @param  string str    string to match (Either SLOOPER_SHAD or SLOOPEE_SHAD)
+ * @return   array of matching shadow(s)
+ */
+protected object * query_sloop_shadows( object ob, string str ) {
+    object * shadow_ob = shadow_list( ob );
+    return filter( shadow_ob, (: return ( program_name( $1 ) == $2 ); :), str );
+}
+
+/**
+ * Calls query_sloop_shadows passing SLOOPER_SHAD
+ * @param  object ob            Object to check for SLOOPER_SHAD
+ * @return        Filtered object from query_sloop_shadows()
+ */
+object find_slooper_shadow( object ob ) {
```

Comment:
confused about the return type of this function
query_sloop_shadows() returns an object array, not an object. 
There's also no return statement in this function, so right now it's not returning anything.